packages/next/src/lib/metadata/get-metadata-route.ts

```diff
@@ -11,14 +11,33 @@ import { normalizePathSep } from '../../shared/lib/page-path/normalize-path-sep'
  * Give it a unique hash suffix to avoid conflicts
  *
  * e.g.
- * /app/opengraph-image.tsx -> /opengraph-image
- * /app/(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ * /opengraph-image -> /opengraph-image
+ * /(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ *
+ * Sitemap is an exception, it should not have a suffix.
+ * As the generated urls are for indexer and usually one sitemap contains all the urls of the sub routes.
+ * The sitemap should be unique in each level and not have a suffix.
+ *
+ * /sitemap -> /sitemap
+ * /(post)/sitemap -> /sitemap
  */
 function getMetadataRouteSuffix(page: string) {
-  let suffix = ''
+  // Remove the last segment and get the parent pathname
+  // e.g. /parent/a/b/c -> /parent
+  // e.g. /parent/opengraph-image.tsx -> /parent
+  const parentPathname = page.slice(0, -(path.basename(page).length + 1))
+  // Only apply suffix to metadata routes except for sitemaps
+  if (page.endsWith('/sitemap')) {
+    return ''
+  }
 
-  if ((page.includes('(') && page.includes(')')) || page.includes('@')) {
-    suffix = djb2Hash(page).toString(36).slice(0, 6)
+  // Calculate the hash suffix based on the parent path
+  let suffix = ''
+  if (
+    (parentPathname.includes('(') && parentPathname.includes(')')) ||
```

Comment:
This should probably be using the [`isGroupSegment`](https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/segment.ts#L3-L6) util

Reply:
This is bit different from `isGroupSegment` but more like `hasGroupSegment` since we're asserting on the whole pathname. `isGroupSegment` only checks if the 1st letter has `(`